### PR TITLE
BaSP-MR-73: Fix POST and PUT, add QA corrections

### PR DESCRIPTION
**Pull request by SantiH97**
Complete code overhaul

### CODE REVIEW

File: src/Components/Classes/index.jsx [hunk 1]
```diff
@@ -1,127 +1,207 @@
...
         }
       });
-
-      const { message, data, error } = await newClass.json();
-      alert(message);
-
-      if (!error) {
-        setClasses([...classes, data]);
-      }
+      console.log(classes);
```
Comment (DelfiLT) [inline src/Components/Classes/index.jsx:73]: Remove console.log

File: src/Components/Classes/index.jsx [hunk 2]
```diff
@@ -1,127 +1,207 @@
...
-  const deleteItem = async (_id) => {
+  const deleteClass = async (_id) => {
     await fetch(`${process.env.REACT_APP_API_URL}/api/classes/${_id}`, {
       method: 'DELETE'
     });
-    const klass = classes.filter((klass) => klass._id === _id);
-    alert(`Class ${klass[0].name} was deleted`);
-    setClasses([...classes.filter((klass) => klass._id !== _id)]);
+    const justOne = classes.filter((justOne) => justOne._id === _id);
+    console.log(justOne);
```
Comment (DelfiLT) [inline src/Components/Classes/index.jsx:86]: Remove console.log

File: src/Components/Classes/index.jsx [hunk 3]
```diff
@@ -1,127 +1,207 @@
...
 
+  const onSubmitToAdd = (e) => {
+    e.preventDefault();
+    console.log(readyToSendAndAdd);
+    addClass(readyToSendAndAdd);
+  };
+  const onSubmitToEdit = (e) => {
+    e.preventDefault();
+    console.log(readyToSendAndEdit);
+    updateClass(currentId, readyToSendAndEdit);
```
Comment (DelfiLT) [inline src/Components/Classes/index.jsx:118]: Remove console.log

**Comment by SantiH97** [comment]
console.logs removed

File: src/Components/Classes/index.jsx [hunk 4]
```diff
@@ -1,127 +1,203 @@
...
         }
       });
-
-      const { message, data, error } = await newClass.json();
-      alert(message);
-
-      if (!error) {
-        setClasses([...classes, data]);
-      }
+      const data = await res.json();
```
Comment (ukiweskamp) [inline src/Components/Classes/index.jsx:73]: if here you try something like this:

        const newClass = {
        _id: data.data._id,
        activityId: {
            name: classes.find((item) => item._id === data.data.classId).activityId?.name
          },
        day: data.data.day,
        hour: data.data.hour,
        slots: data.data.slots
      };


      setClasses([...classes, newClass]);
    } catch (error) {
      console.error(error);
      
 i think it should work

Comment (SantiH97) [reply src/Components/Classes/index.jsx:73]: Even with that implementation the code behaves the same

**Review by ukiweskamp — APPROVED** [review]

**Review by Dannylez — APPROVED** [review]

**Comment by SantiH97** [comment]
[Classes - BUG](https://trello.com/c/reFFikxV/73-classes-bug)

**Pull request merged**